arangod/Replication2/ReplicatedLog/Algorithms.cpp

```diff
@@ -589,21 +618,23 @@ auto algorithms::calculateCommitIndex(
   // this certainly means we could not reach a quorum;
   // indexes cannot be empty because this particular case would've been handled
   // above by comparing actualWriteConcern to 0;
-  CommitFailReason::NonEligibleServerRequiredForQuorum::CandidateMap candidates;
+  CommitDetails::NonEligibleServerRequiredForQuorum::CandidateMap candidates;
   for (auto const& p : indexes) {
     if (p.isFailed()) {
       candidates.emplace(
-          p.id, CommitFailReason::NonEligibleServerRequiredForQuorum::kFailed);
+          p.id, CommitDetails::NonEligibleServerRequiredForQuorum::kFailed);
     } else if (p.isExcluded()) {
       candidates.emplace(
-          p.id,
-          CommitFailReason::NonEligibleServerRequiredForQuorum::kExcluded);
+          p.id, CommitDetails::NonEligibleServerRequiredForQuorum::kExcluded);
+    } else if (p.lastTerm() != lastTermIndex.term) {
+      candidates.emplace(
+          p.id, CommitDetails::NonEligibleServerRequiredForQuorum::kWrongTerm);
```

Comment:
`WrongTerm` might be misleading, because the follower might be in the right term, it just never has acked an entry of the current term. We can leave it as it is now and I create a ticket to sort this out later.